thavalon-server/src/connections/game_handlers.rs

```diff
@@ -262,6 +274,39 @@ async fn client_connection(socket: WebSocket, client_id: String, mut lobby_chann
     }
 }
 
+/// Helper function for monitoring a lobby, intended to run as a tokio task. This will remove the lobby from
+/// GameCollection once the lobby ends or exceeds the maximum lobby lifetime.
+async fn monitor_lobby_task(mut lobby_channel: LobbyChannel, friend_code: String, game_collection: GameCollection) {
+    // Lobby timeout is 6 hours from creation across all phases.
+    let timeout = tokio::time::delay_until(Instant::now() + Duration::from_secs(60 * 60 * 6));
+    let mut timeout_channel = lobby_channel.clone();
+    tokio::select! {
+        _ = timeout => {
+            log::error!("Lobby {} has exceeded timeout, killing this lobby now.", &friend_code);
+            timeout_channel.send((LobbyCommand::EndGame, None)).await;
+        }
+        _ = monitor_game_for_completion(lobby_channel) => {
+            log::info!("Lobby {} completed, removing it from game collection.", &friend_code);
+        }
+    }
+    game_collection.lock().unwrap().remove(&friend_code);
+}
+
+async fn monitor_game_for_completion(mut lobby_channel: LobbyChannel) {
```

Comment:
Instead of polling, a pretty common pattern I've seen is to have a `tokio::sync::oneshot` channel that's triggered once the task ends. I think publishing to the channel from `end_game` and then awaiting it in the `select` here would have the same effect as polling, but complete as soon as the game ends and avoid the extra round of messages.

Reply:
makes sense. that definitely seems cleaner than polling and relying on the receiver to drop when a game ends, though it's one extra channel to set up. switched to that approach